### PR TITLE
feat: add `assert/is-complex-string`

### LINKED ISSUE #1370: [RFC]: add `@stdlib/assert/is-complex-string`
### Description

This RFC proposes adding package `@stdlib/assert/is-complex-string`.
 
The package should be similar in structure to [`@stdlib/complex/reviver-float32`](https://github.com/stdlib-js/stdlib/tree/64931c5fdb4abc3a339b619658d3b5acc089e556/lib/node_modules/%40stdlib/complex/reviver-float32).

The goal is to assert if an input string is of a complex number format string`(a + ib)` or `(a+bi)`
where, a and b are numbers 

```js
 function isComplexString( str ) => boolean
```

Eg : isComplexString( "32+12i" ) returns true
Eg : isComplexString( "32+i12" ) returns true
Eg : isComplexString( null ) returns true
Eg : isComplexString( "" ) returns false
Eg : isComplexString( "abcd+12i" ) returns false
Eg : isComplexString( Infinity ) returns false
 
> Package: `@stdlib/assert/is-complex-string` Alias: `isComplexString`

### Related Issues

Can help with issues #1332 and #1333

### Questions

This is my first issue so I'd like to know if the way I have proposed the feature is correct

### Other

_No response_

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by marsian83**
Resolves #1370

## Description
This PR adds the package @stdlib/assert/is-complex-string

The package implements assertion for a string represeting a complex number.

Alias: isComplexString

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1370

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by marsian83** [comment]
please review @Planeshifter

File: lib/node_modules/@stdlib/assert/is-complex-string/README.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# isComplexString
+
+> Tests whether input string is a complex number representation`.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-complex-string/README.md:23]:
```suggestion
> Tests whether input string is a complex number representation.
```

File: lib/node_modules/@stdlib/assert/is-complex-string/README.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+## Usage
+
+```js
+var isComplexString = require( '@stdlib/assert/is-complex-string' );
+```
+
+#### isComplexString( str )
+
+Tests whether input string is a complex number representation`.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-complex-string/README.md:45]:
```suggestion
Tests whether input string is a complex number representation.
```

File: lib/node_modules/@stdlib/assert/is-complex-string/README.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+<!-- /.intro -->
+
+<!-- Package usage documentation. -->
+
+<section class="usage">
+
+## Usage
+
+```js
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-complex-string/README.md:39]:
```suggestion
```javascript
```

File: lib/node_modules/@stdlib/assert/is-complex-string/README.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+var isComplexString = require( '@stdlib/assert/is-complex-string' );
+```
+
+#### isComplexString( str )
+
+Tests whether input string is a complex number representation`.
+
+```js
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-complex-string/README.md:47]:
```suggestion
```javascript
```

File: lib/node_modules/@stdlib/assert/is-complex-string/README.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+<!-- Package usage examples. -->
+
+<section class="examples">
+
+## Examples
+
+<!-- eslint no-undef: "error" -->
+
+```js
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-complex-string/README.md:76]: same comment

File: lib/node_modules/@stdlib/assert/is-complex-string/README.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+<section class="references">
+
+</section>
+
+<!-- /.references -->
+
+<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->
+
+<section class="related">
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-complex-string/README.md:115]: You are missing section `See Also`, please add it

File: lib/node_modules/@stdlib/assert/is-complex-string/docs/repl.txt [hunk 7]
```diff
@@ -0,0 +1,19 @@
+{{alias}}( str )
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-complex-string/docs/repl.txt:1]:
```suggestion

{{alias}}( str )
```

File: lib/node_modules/@stdlib/assert/is-complex-string/docs/repl.txt [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+    Returns
+    -------
+    out: boolean
+
+    Examples
+    --------
+    > var str = '4 + 9i';
+    > var b = {{alias}}( str )
+
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-complex-string/docs/repl.txt:16]: Add return type here

File: lib/node_modules/@stdlib/assert/is-complex-string/docs/types/index.d.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+* @param str
+* @returns boolean
+*
+* @example
+* var isComplexString = require( '@stdlib/assert/is-complex-string' );
+*
+* var str = '4 + 6i';
+*
+* var b = isComplexString( str );
+* returns true
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-complex-string/docs/types/index.d.ts:33]:
```suggestion
* // returns true
```

File: lib/node_modules/@stdlib/assert/is-complex-string/docs/types/test.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+{
+	const str = '54 + 3i';
+	isComplexString( str ); // $ExpectType boolean
+}
+
+// The compiler throws an error if the function is provided insufficient arguments...
+{
+	isComplexString(); // $ExpectError
+	isComplexString( 'beep', 'boop' ); // $ExpectError
+}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-complex-string/docs/types/test.ts:34]: more tests can be added

File: lib/node_modules/@stdlib/assert/is-complex-string/lib/main.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isString = require( '@stdlib/assert/is-string' );
+var parse = require( '@stdlib/complex/base/parse');
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-complex-string/lib/main.js:24]:
```suggestion
var parse = require( '@stdlib/complex/base/parse' );
```

File: lib/node_modules/@stdlib/assert/is-complex-string/lib/main.js [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+
+
+// MAIN //
+
+/**
+* Test whether input string is a complex number representation.
+*
+* @param {string} str - input string
+* @returns {boolean} a boolean indicated if the string is a valid complex number representation.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-complex-string/lib/main.js:33]:
```suggestion
* @returns {boolean} a boolean indicated if the string is a valid complex number representation
```

File: lib/node_modules/@stdlib/assert/is-complex-string/lib/main.js [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+*
+* @example
+* var str = '4 + 6i';
+*
+* var isCmplx = isComplexString( str );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-complex-string/lib/main.js:38]:
```suggestion
* var v = isComplexString( str );
```

File: lib/node_modules/@stdlib/assert/is-complex-string/test/test.js [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+		'1.523 - 4.234i',
+		'-1.5 + 4.2i',
+		'1.0e10 -2.0e-10i',
+		'1.0E10 + 4.1e+10i',
+		'1+2i',
+		'1-0i',
+		'0+1i',
+		'-0.0 + 2.0i',
+		'-Infinity + 5i',
+
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-complex-string/test/test.js:53]:
```suggestion
```

File: lib/node_modules/@stdlib/assert/is-complex-string/test/test.js [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+		'Infinity-Infinityi'
+
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/assert/is-complex-string/test/test.js:55]:
```suggestion
```

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Requested changes, once done this PR can be reviewed again

**Pull request closed without merging**